services/app-api/src/postgres/contractAndRates/updateDraftContract.ts

```diff
@@ -66,62 +68,82 @@ async function updateDraftContract(
             return new NotFoundError(err)
         }
 
-        await client.contractRevisionTable.update({
-            where: {
-                id: currentRev.id,
-            },
-            data: {
-                populationCovered: populationCovered,
-                programIDs: programIDs,
-                riskBasedContract: riskBasedContract,
-                submissionType: submissionType,
-                submissionDescription: submissionDescription,
-                contractType: contractType,
-                contractExecutionStatus,
-                contractDocuments: {
-                    create: contractDocuments,
+        await client.$transaction([
```

Comment:
I assume this works, client.transaction with an array of client.whatever commands. When I did it before I got a transaction handle that I then called

Reply:
Yes this is [sequential cal.](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#sequential-prisma-client-operations). I believe that's what we want when making simple sequential requests like this. The syntax I saw in your work looked like the [interactive transation syntax](https://www.prisma.io/docs/concepts/components/prisma-client/transactions#interactive-transactions). Don't need that here.

Reply:
Should we delete all the records or [disconnect](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#disconnect) them? <strike>I would think deleting would cause some issues with revisions. If an old revision still had these relationships, I don't think deleting would work; if it did, those revisions would then have a broken relationship. I think testing around this would be good. </strike>

Ah nvm. Looks like in `unlockContract.ts`, we don't have it implemented fully to copy over all the data to the new revision. Also, in the Prisma schema, documents and revisions have a one-to-many relationship, so a document could not be related to many contract revisions. So new revisions just create a whole new set of documents?

Reply:
I wonder if v2 of document and contact updating could be something like this:

1.  Programmatically compare updated data with db data and split data into three buckets: create, disconnect, and update.
2. Then we use these buckets in their own Prisma operation in the sequential operation: 
   - createMany, 
   - disconnectMany ( this one can be in `contractRevisionTable.update`) or deleteMany
   - updateMany
3. This would have to be repeated for every document and contact table 🤡. 

Just deleting and recreating seems wrong, I know we don't use the `updatedAt` or `createdAt` fields, but doing this would make those fields a field of LIES. (Not a strong feeling, but a medium one)

Reply:
I wonder if v2 of document and contact updating could be something like this:

Programmatically compare updated data with db data and split data into three buckets: create, disconnect/delete, and update.
Then we use these buckets in their own Prisma operation in the sequential operation:
createMany,
disconnectMany ( this one can be in contractRevisionTable.update) or deleteMany
updateMany
This would have to be repeated for every document and contact table 🤡.
Just deleting and recreating seems wrong, I know we don't use the updatedAt or createdAt fields, but doing this would make those fields a field of LIES. (Not a strong feeling, but a medium one)

Reply:
@JasonLin0991 Yep. I hear you. You are rehasing the conversation MacRae and I had yesterday. I had functions that covered each table. What you are proposing is slightly different but goes down a similar path of multi part update where we create (I was going to create an update in each table) and then disconnect/delete. I'm down to do this but I would like to discussion as a team first to all get on same page.

Reply:
@JasonLin0991 I'm gonna open on a follow on PR to allow us to discuss this  - we can continue on shipping other work but use that PR to iterate on a v2 if we want. 

 Will merge this work as is.